sycl/unittests/kernel-and-program/KernelBundle.cpp

```diff
@@ -0,0 +1,29 @@
+//==- KernelBuildOptions.cpp - Kernel build options processing unit test --==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===---------------------------------------------------------------------===//
+
+#include <gtest/gtest.h>
+#include <helpers/PiMock.hpp>
+#include <sycl/sycl.hpp>
+
+class KernelA;
+
+TEST(KernelBundle, HasKernel) {
+  sycl::unittest::PiMock Mock;
+  sycl::platform Plt = Mock.getPlatform();
+  
+  const sycl::device Dev = Plt.get_devices()[0];
+
+  sycl::queue Queue{Dev};
+
+  const sycl::context Ctx = Queue.get_context();
+
+  sycl::kernel_bundle KernelBundle =
+    sycl::get_kernel_bundle<KernelA, sycl::bundle_state::executable>(Ctx);
+
+  EXPECT_TRUE(KernelBundle.has_kernel<KernelA>());
+}
```

Comment:
There is already a KernelBundle.cpp unittest in sycl/unittests/SYCL2020/KernelBundle.cpp. Would it make sense to move this new test case to it?

Reply:
Moved to KernelID.cpp as an additional test case. Will add one for invalid KernelName once the other conversation about noexcept has been resolved.